src/main/kotlin/com/github/igorperikov/mightywatcher/Launcher.kt

```diff
@@ -14,29 +26,64 @@ const val PARALLELISM_LEVEL_ENV_NAME = "PARALLELISM"
 const val DEFAULT_PARALLELISM_LEVEL = "10"
 const val DAYS_SINCE_LAST_UPDATE_ENV_NAME = "DAYS"
 const val DEFAULT_DAYS_SINCE_LAST_UPDATE = "90"
+const val TOKEN_ENV_NAME = "TOKEN"
 
 object Launcher {
     @JvmStatic
     private val log = LoggerFactory.getLogger(this.javaClass)
 
-    private val githubService = GithubService(
-        RestGithubApiClient.fromEnv(),
-        System.getenv(INCLUDE_LANG_ENV_NAME)?.split(",")?.toHashSet() ?: setOf(),
-        System.getenv(EXCLUDE_REPOS_ENV_NAME)?.split(",")?.toHashSet() ?: setOf()
-    )
-    private val importService = ImportService(
-        githubService,
-        LabelsService(githubService, EasyLabelsStorage()),
-        parallelismLevel = (System.getenv(PARALLELISM_LEVEL_ENV_NAME) ?: DEFAULT_PARALLELISM_LEVEL).toInt(),
-        daysInPast = (System.getenv(DAYS_SINCE_LAST_UPDATE_ENV_NAME) ?: DEFAULT_DAYS_SINCE_LAST_UPDATE).toLong()
-    )
-    private val groupingService = GroupingService.withDefaultTimeGroups()
+    fun initHttpClient(githubToken: String): HttpClient {
+        return HttpClient(Apache) {
+            install(JsonFeature) {
+                serializer = JacksonSerializer {
+                    findAndRegisterModules()
+                    registerModule(JavaTimeModule())
+                }
+            }
+
+            defaultRequest {
+                url {
+                    protocol = URLProtocol.HTTPS
+                    host = "api.github.com"
+                }
+                header("Accept", "application/vnd.github.v3+json")
+                header("User-Agent", "IgorPerikov/mighty-watcher")
+                header("Authorization", "token $githubToken")
+            }
+        }
+    }
+
+    val applicationModule = module {
+        single { initHttpClient(getProperty(TOKEN_ENV_NAME)) }
```

Comment:
will it complain, if env variable wasn't set?

Reply:
Yes, if the property is not set it will throw `MissingPropertyException("Property '$key' not found")`